plugins/outputs/wavefront/wavefront.go

```diff
@@ -97,9 +100,21 @@ func (w *Wavefront) Connect() error {
 		connectionURL = senderURLFromHostAndPort(w.Host, w.Port)
 	}
 
+	tlsConfig, err := w.TLSConfig()
+
+	if err != nil {
+		return err
+	}
+
+	if w.Timeout == 0 {
+		w.Timeout = config.Duration(10 * time.Second)
```

Comment:
now we have 3 different timeout values :)

* sample config says 3s
* httpConfig uses 5s
* wavefront defaults to 10s

If you want wavefront to use 10s by default that is fine, but the sample config needs to line up with that.

I would also handle setting default values during `init()` as you have already done.

Reply:
I didn't embed `httpConfig` because we don't support support overriding the http client to that degree. I'd like to look into it for the future.

I appreciate the callout about `init` - I hadn't looked closely at the lifecycle and I was concerned that if there was no `toml` value for the timeout, we'd get zero (since we don't get the `httpConfig` default). But after playing with it and adding debugging logs here and there, I now understand that the defaults set in `init`, and we don't need line `109` - so I'll take that out. 👍